### PR TITLE
tso/local: remove local tso completely

### PR DESCRIPTION
**Pull request by okJiang**
### What problem does this PR solve?

Issue Number: Close https://github.com/tikv/pd/issues/8802
### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/mcs/tso/server/server.go [hunk 1]
```diff
@@ -272,15 +272,6 @@ func (s *Server) GetTSOAllocatorManager(keyspaceGroupID uint32) (*tso.AllocatorM
 	return s.keyspaceGroupManager.GetAllocatorManager(keyspaceGroupID)
 }
 
-// IsLocalRequest checks if the forwarded host is the current host
-func (*Server) IsLocalRequest(forwardedHost string) bool {
```
Comment (rleungx) [inline pkg/mcs/tso/server/server.go:276]: Why remove it?

File: pkg/tso/allocator_manager.go [hunk 2]
```diff
@@ -217,17 +214,6 @@ func (am *AllocatorManager) getGroupIDStr() string {
 	return strconv.FormatUint(uint64(am.kgID), 10)
 }
 
-// GetTimestampPath returns the timestamp path in etcd.
-func (am *AllocatorManager) GetTimestampPath() string {
```
Comment (rleungx) [inline pkg/tso/allocator_manager.go:221]: ditto

File: pkg/mcs/tso/server/server.go (hunk 1 above)
Comment (okJiang) [reply pkg/mcs/tso/server/server.go:276]: It is unused🤔

File: pkg/tso/allocator_manager.go (hunk 2 above)
Comment (okJiang) [reply pkg/tso/allocator_manager.go:221]: `GetTimestampPath` is used to print log when `created allocator manager`. I think 

```go
log.Info("created allocator manager",
		zap.Uint32("keyspace-group-id", group.ID))
```

is enough, do not have to know the TimestampPath.

Comment (rleungx) [reply pkg/tso/allocator_manager.go:221]: The comment is important.

Comment (rleungx) [reply pkg/tso/allocator_manager.go:221]: Got it. BTW, I found we still have some comments mentioning `KeyspaceGroupLocalTSPath`, please also remove it.

**Review by rleungx — APPROVED** [review]

File: client/client.go [hunk 3]
```diff
@@ -568,7 +568,7 @@ func (c *client) GetTSAsync(ctx context.Context) tso.TSFuture {
 
 // GetLocalTSAsync implements the TSOClient interface.
 //
-// Deprecated: Local TSO will be completely removed in the future. Currently, regardless of the
+// Deprecated: Currently, regardless of the
 // parameters passed in, this method will default to returning the global TSO.
```
Comment (JmPotato) [inline client/client.go:572]:
```suggestion
// Deprecated: the Local TSO feature has been deprecated. Regardless of the parameters passed, the behavior of this interface will be equivalent to `GetTSAsync/GetTS`. If you want to use a separately deployed TSO service, please refer to the deployment of the TSO microservice.
```

File: client/client.go [hunk 4]
```diff
@@ -582,7 +582,7 @@ func (c *client) GetTS(ctx context.Context) (physical int64, logical int64, err
 
 // GetLocalTS implements the TSOClient interface.
 //
-// Deprecated: Local TSO will be completely removed in the future. Currently, regardless of the
+// Deprecated: Currently, regardless of the
```
Comment (JmPotato) [inline client/client.go:585]: Ditto.

**Review by JmPotato — APPROVED** [review]

**Review by niubell — APPROVED** [review]

**Pull request merged**